hecate_lexer/src/token.rs

```diff
@@ -1,9 +1,8 @@
-use hecate_util::ast::{BinaryOp, UnaryOp};
-
-pub enum Token {
-    Identifier,
-    BinaryOp(BinaryOp),
-    UnaryOp(UnaryOp),
-    AssignmentOp,
-    Delimiter,
+#[derive(Debug, PartialEq, Eq)]
+pub enum Token<'a> {
+    Identifier(&'a str),
+    ControlCharacter(char),
```

Comment:
1. Maybe just name "Control" (bit of a long name)?
2. Errors for control tokens which have to be together: `a + = b;;` vs `a += b`. We can also not just join all Control together; `let a =-b` is valid and would require 2 separate tokens, aswell as other ambiguities (like generics or braces).

Until now i usually added a bool to indicate followed by control vs (anything else or whitespace). 
bool followed_by_control:
- `+=` true
- `+ =` false
- `+4` false
- `+a` false
- `-(` true
The logic for "followed_by_whitespace" is similar and depends on preference.

i am not particularily happy with that solution so if theres a better one i would be happy to use that

Reply:
Regarding naming, what about `ControlChar`, `Control` conveys not enough information in my opinion.

I am not sure if this solution is any better than the original `Operator` approach.